src/topojson/topojson.ml

```diff
@@ -698,24 +696,23 @@ module Make (J : Intf.Json) = struct
     let junction_table : (Position.t, Set.t) Hashtbl.t =
       Hashtbl.create (List.length lines)
     in
-    let junction : Position.t list = [] in
+    let junction : Position.t list ref = ref [] in
     List.iter
       (fun line ->
         let line_length = Array.length line in
-        for i = 1 to line_length - 2 do
+        for i = 0 to line_length - 1 do
           let point1 = line.(i) in
-          let point2 = line.(i - 1) in
-          let point3 = line.(i + 1) in
+          let point2 = if i > 0 then line.(i - 1) else line.(line_length - 1) in
+          let point3 = if i < line_length - 1 then line.(i + 1) else line.(0) in
           if Hashtbl.mem junction_table point1 then
             let neighbors = Hashtbl.find junction_table point1 in
             Hashtbl.replace junction_table point1
               (Set.add point2 (Set.add point3 neighbors))
           else
             Hashtbl.add junction_table point1 (Set.of_list [ point2; point3 ]);
           let neighbors = Hashtbl.find junction_table point1 in
-          if Set.cardinal neighbors > 2 then
-            ref junction := point1 :: !(ref junction)
+          if Set.cardinal neighbors > 2 then junction := point1 :: !junction
```

Comment:
With the above changes this will become `List.length` when we store the list of possible neighbours and store a new pair _iff_ they are different as defined in the algorithm.